src/SDKs/Azure.Base/data-plane/Azure.Base/Pipeline/ResponseClassifier.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+
+namespace Azure.Base.Pipeline
+{
+    public abstract class ResponseClassifier
+    {
+        /// <summary>
+        /// Specifies if the response should terminate the pipeline and not be retried
+        /// </summary>
+        public abstract bool IsFatalErrorResponse(HttpPipelineResponse pipelineResponse);
+
+        /// <summary>
+        /// Specifies if the exception should terminate the pipeline and not be retried
+        /// </summary>
+        public abstract bool IsFatalException(Exception exception);
+
+        /// <summary>
+        /// Specifies if the response is not successful but can be retried
+        /// </summary>
+        public abstract bool IsErrorResponse(HttpPipelineResponse pipelineResponse);
+    }
+
+    public class DefaultResponseClassifier : ResponseClassifier
+    {
+        public static DefaultResponseClassifier Singleton { get; } = new DefaultResponseClassifier();
+
+        public override bool IsFatalErrorResponse(HttpPipelineResponse pipelineResponse)
+        {
```

Comment:
I would think that, by default, we'd want to include most of the `HTTP 400` family here;  outside of a `429 where we'd need to apply a throttle and a `404` when you have a known race condition on resource creating, retrying a client error without making changes to the request should result in continued failures for a sane system.

Reply:
Yeah, sounds reasonable.

Reply:
I think the default should be very conservative about what it retries.  Everything should be fatal except a small list of codes, possibly just 429 and 503 to start?

Reply:
Done